tests/TopDownProteomics.Tests/ProForma/ProFormaParserTests.cs

```diff
@@ -219,6 +219,88 @@ public void Rule6_7_Invalid(string proFormaString)
             Assert.Throws<ProFormaParseException>(() => _parser.ParseString(proFormaString));
         }
 
+        public void PossibleSiteAmbiguityRules()
+        {
+            const string proFormaString = "PROT[Phospho|#eg]EOS[#eg]FORMS[#eg]";
+            var term = _parser.ParseString(proFormaString);
+
+            Assert.AreEqual("PROTEOSFORMS", term.Sequence);
+            Assert.IsNotNull(term.Tags);
+            Assert.AreEqual(3, term.Tags.Count);
+            Assert.IsNull(term.NTerminalDescriptors);
+            Assert.IsNull(term.CTerminalDescriptors);
+
+            ProFormaTag tag1 = term.Tags[0];
+            Assert.AreEqual(3, tag1.ZeroBasedIndex);
+            Assert.AreEqual(2, tag1.Descriptors.Count);
+            Assert.AreEqual(1, tag1.Descriptors.OfType<ProFormaAmbiguityDescriptor>().Count());
+            Assert.AreEqual(ProFormaAmbiguityAffix.PossibleSite, tag1.Descriptors.OfType<ProFormaAmbiguityDescriptor>().Single().Affix);
+            Assert.AreEqual("eg", tag1.Descriptors.OfType<ProFormaAmbiguityDescriptor>().Single().Group);
+
+            ProFormaTag tag2 = term.Tags[1];
+            Assert.AreEqual(6, tag2.ZeroBasedIndex);
+            Assert.AreEqual(1, tag2.Descriptors.Count);
+            Assert.AreEqual(ProFormaAmbiguityAffix.PossibleSite, tag2.Descriptors.Single().Key);
+            Assert.AreEqual(ProFormaAmbiguityAffix.PossibleSite, (tag2.Descriptors.Single() as ProFormaAmbiguityDescriptor).Affix);
+            Assert.AreEqual("eg", tag2.Descriptors.Single().Value);
+            Assert.AreEqual("eg", (tag2.Descriptors.Single() as ProFormaAmbiguityDescriptor).Group);
+        }
+
+        public void RangeAmbiguityRules()
+        {
+            const string proFormaString = "PROT[mass:19|A->]EOSFORMS[<-A]";
```

Comment:
I think you're right, it does come back to the left to right issue. So, for ProForma, that should be second nature to assume that the T is included. While I think the notation is a bit misleading, it is logical .... and I can't think of anything better! Let's roll with this for now and keep thinking before we lock it in with another publication.

Reply:
What about `PROT[mass:19|<-group]EOSFORMS[<-group]`?

Reply:
It seems less like a range, but it points to the limits.
`PR[Methyl|<-group2]OT[mass:19|<-group1]EOSFOR[<-group2]MS[<-group1]`

Reply:
The nested case really motivates the arrows, though...
`PR[Methyl|group2->]OT[mass:19|group1->]EOSFOR[<-group2]MS[<-group1]`

Reply:
For what it's worth, we have agreement here (right-centric view of mods) that it would be confusing to have the arrows not include the start of the range.

Reply:
I'll keep it in for now, but we should keep thinking about this...